skimage/measure/fit.py

```diff
@@ -59,7 +113,57 @@ def _warn_or_msg(msg, warn_only=True):
     return None
 
 
-class LineModelND(BaseModel):
+def _deprecate_no_args(cls):
+    """Class decorator to allow, deprecate no input arguments to ``__init__``.
+
+    Makes a new ``__init__`` method, that a) will allow option of passing no
+    arguments, and b) when used thus, raises a deprecation warning.  Otherwise
+    defers to an assumed-existing ``_args_init`` instance method to deal with
+    input arguments.  If there are no parameters, set desired parameters to
+    None, to signal uninitialized object.
+
+    At the end of deprecation we can drop this decorator, and rename
+    ``_args_init`` to ``__init__``.
+    """
+
+    args_init_sig = inspect.signature(cls._args_init)
+    cls._init_args = [k for k in args_init_sig.parameters if k != 'self']
+
+    def init(self, *args, **kwargs):
+        if len(args) or len(kwargs):
+            self._args_init(*args, **kwargs)
+            return
+        warn(
+            f'Calling ``{cls.__name__}()`` (without arguments) has been '
+            f'deprecated since version {_PARAMS_DEP_START} and will be '
+            f'removed in version {_PARAMS_DEP_STOP}; see help for '
+            f'``{cls.__name__}``.',
+            category=FutureWarning,
+            stacklevel=2,
+        )
+        # Blank initialization.
+        for k in cls._init_args:
+            setattr(self, k, None)
+
+    init.__signature__ = args_init_sig
+    cls.__init__ = init
+    return cls
+
+
+def _deprecate_model_params(func):
+    """Deprecate `params` argument of various model methods."""
+    func = deprecate_parameter(
+        'params',
+        start_version=_PARAMS_DEP_START,
+        stop_version=_PARAMS_DEP_STOP,
+        modify_docstring=False,
+    )(func)
+    func.__doc__ = func.__doc__.replace('{{ start_version }}', _PARAMS_DEP_START)
+    return func
+
+
+@_deprecate_no_args
+class LineModelND(_BaseModel):
```

Comment:
I'm wondering how the interaction between these three parts affects users that subclass `LineModelND`. E.g. will they get a useful warning and do we need to worry about this?